compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/PackageDependencyGraphBuilder.java

```diff
@@ -225,22 +270,46 @@ private NodeStatus addNewVertex(Vertex vertex, DependencyNode newPkgDep, boolean
 
         // There exists another version in the graph.
         DependencyNode existingPkgDep = vertices.get(vertex);
-        PackageDescriptor resolvedPkgDesc = handleDependencyConflict(newPkgDep, existingPkgDep);
-
-        // If the existing dependency scope is DEFAULT, use it. Otherwise use the new dependency scope.
-        PackageDependencyScope depScope =
-                existingPkgDep.scope() == PackageDependencyScope.DEFAULT ?
-                        PackageDependencyScope.DEFAULT :
-                        newPkgDep.scope();
-
-        // If the existing dependency scope is DEFAULT, use it. Otherwise use the new dependency scope.
-        DependencyResolutionType resolutionType =
-                existingPkgDep.resolutionType() == DependencyResolutionType.SOURCE ?
-                        DependencyResolutionType.SOURCE :
-                        newPkgDep.resolutionType();
-
-        DependencyNode resolvedPkgDep = new DependencyNode(resolvedPkgDesc, depScope, resolutionType);
-        NodeStatus nodeStatus = getNodeStatus(vertex, existingPkgDep, newPkgDep, unresolved);
+
+        // If the existing dependency is an error node, we continue with the error node.
+        if (existingPkgDep.errorNode()) {
+            return NodeStatus.REJECTED;
+        }
+
+        DependencyNode resolvedPkgDep;
+        NodeStatus nodeStatus;
+
+        if (newPkgDep.errorNode()) {
+            resolvedPkgDep = newPkgDep;
+            nodeStatus = NodeStatus.ACCEPTED;
```

Comment:
The scenario that meets this condition:
* App -->bar (1.0.0) --> foo (0.6.0) - All version are locked in the dependencies.toml
* User specifies foo (0.7.0) is Ballerina.toml
Refer case-0004 for the testcase.

Implementation:
* Create BlendedManifest and capture error nodes.
    - foo(0.6.0) will be added to the depContainer with `Origin.Error`
* Populate static dependency graph
    - This will add bar (1.0.0) and foo (0.6.0) to the dependency graph. At this point, the dependency tree is blindly generated using the static graph of `bar`. 
    Hence, no error nodes will be captured. When adding to the graph, the direct deps are added as resolved nodes and the transitive nodes are added as unresolved.
* Update dependency versions 
    - Check unresolved nodes and try to resolve them. At this point, the unresolved foo (0.6.0) (existing) will be compared with the error node in the BlendedManifest (new node).